src/sysext/sysext.c

```diff
@@ -91,6 +98,66 @@ static const struct {
         }
 };
 
+static bool need_reload(void) {
```

Comment:
this can fail for so many reasons, please propagate an error here, i.e change return value to `int`, so that < 0 is error, == 0 means "no reload needed", and > 0 means "reload please"